state-sync/storage-service/types/src/lib.rs

```diff
@@ -282,51 +282,54 @@ impl ProtocolMetadata {
             | GetStorageServerSummary
             | GetNumberOfAccountsAtVersion(_) => true,
             GetAccountStatesChunkWithProof(request) => {
-                let chunk_size = match CompleteDataRange::new(
-                    request.start_account_index,
-                    request.end_account_index,
-                ) {
-                    Ok(range) => range.len(),
-                    Err(_) => return false,
-                };
-                self.max_account_states_chunk_size >= chunk_size
+                match CompleteDataRange::new(request.start_account_index, request.end_account_index)
```

Comment:
nit: I wonder if we could we use something like `map_or_else()` (twice?) here to avoid the two `false` branches? https://doc.rust-lang.org/std/result/enum.Result.html#method.map_or. 🤔 Feel free to ignore it though if you don't care 😄

Reply:
Updated the code to use `map_or`.

(Note: Thanks for this nit--I care about it. Before the last commit, I'd actually considered between  `match` vs. `map_or` for some time. Eventually I'd just picked one to avoid bothering you with every single small code detail :) . And I figured you'd bring it up anyway it could be done in a better way.)
Please let me know if you're okay with this change. Thanks @JoshLind !

Reply:
Sounds good! Yeah, I think in this case it's borderline/matter of taste. I don't feel super strongly :)